doc/user-guide/MongooseIM-High-level-Architecture.md

```diff
@@ -0,0 +1,59 @@
+## Inside MongooseIM
+
+MongooseIM comes with a large set of modules that sysadmins can configure and potentially enable, modifying the behaviour of MongooseIM. See '[Extension Modules](../advanced-configuration/Modules/)' for more info.
+
+For a reminder, MongooseIM manages two sets of data: transient for session data management, and persistent for archive and configurations.
+
+## MongooseIM and its ecosystem in a datacenter
+
+### Databases
+
+Please refer to '[Database Backends](../advanced-configuration/database-backends-configuration/)' doc for more configuration information.
+
+* Transient databases
+
+Mnesia will run on the same nodes as MongooseIM.
+
+Redis will be a separate cluster, not using the same nodes as MongooseIM.
+
+* Persistant databases
+
+Both RDBMS (SQL) and NOSQL (Riak KV, Cassandra) 
+
+* LDAP
+
+LDAP will also run on a separate cluster.
+
+### Frontend
+
+Native clients, on platforms such as Android, iOS, Windows, Linux, macOS, will preferrably use a plain XMPP over TCP connections. Web clients cannot use these TCP connections.
+
+Web clients will preferrably use XMPP over websockets, or the now less relevant XMPP over BOSH, using long-lived HTTP connections, more and more used as fallback.
+
+Any client could use the client REST API, which is using HTTP request/responses.
```

Comment:
We don't have it yet :) we can merge this PR only when the functionality is merged.